fv3core/stencils/fxadv.py

```diff
@@ -1,89 +1,107 @@
 from gt4py.gtscript import PARALLEL, computation, horizontal, interval, region
 
 from fv3core.decorators import FrozenStencil
+from fv3core.stencils.d2a2c_vect import contravariant
 from fv3core.utils.grid import GridData, GridIndexing, axis_offsets
 from fv3core.utils.typing import FloatField, FloatFieldIJ
 
 
 # TODO: the mix of local and global regions is strange here
 # it's a workaround to specify DON'T do this calculation if on the tile edge
-def main_ut(
+def main_uc_contra(
     uc: FloatField,
     vc: FloatField,
     cosa_u: FloatFieldIJ,
     rsin_u: FloatFieldIJ,
-    ut: FloatField,
+    uc_contra: FloatField,
 ):
+    """
+    Args:
+        uc: covariant c-grid x-wind (in)
+        vc: covariant c-grid y-wind (in)
+        cosa_u: ??? (in)
+        rsin_u: ??? (in)
```

Comment:
It is certainly hard to maintain copies of the descriptions for arguments used in many places, though I'm also nervous about not documenting arguments. I could make this easier by writing a linting test that reads the docstrings of all functions in `stencils/` and fails if any have the same arguments with different descriptions (ignoring the in/out at the end)? If we move to some kind of system where we're importing grid data instead of passing it as an argument, I think it might be more clear that you can look at some central place to see its description, and also less important for a coder to read if the grid data is automatically supplied (e.g. by our stencil wrapper).

Can we leave this however-it-gets-done for now, and unify the style after discussing what we want as a long term solution? This could happen as part of preparing the code to present at AMS, e.g. if we're going to set up Sphinx documentation.

Reply:
ok sure, you can leave as-is for now.